### ISSUE TITLE
ASPNETCORE_URLS environment variable is ignored by "dotnet run"

### ISSUE THREAD

**Issue by mikeharder** [aspnet/MetaPackages#264]
# Repro Steps
1. `dotnet new web`
2. `set ASPNETCORE_URLS=http://127.0.0.1:0`
3. `dotnet run` (note ASPNETCORE_URLS is ignored)
```
> dotnet run
Using launch settings from D:\Temp\foo2\Properties\launchSettings.json...
Hosting environment: Development
Content root path: D:\Temp\foo2
Now listening on: https://localhost:5001
Now listening on: http://localhost:5000
Application started. Press Ctrl+C to shut down.
```
4. `dotnet run --no-launch-profile` (note ASPNETCORE_URLS is used)
```
> dotnet run --no-launch-profile
Hosting environment: Production
Content root path: D:\Temp\foo2
Now listening on: http://127.0.0.1:50897
Application started. Press Ctrl+C to shut down.
```

**Comment by Tratcher** [comment]
What's in your launchSettings.json? It sounds like its setting ASPNETCORE_URLS for you.

**Comment by JunTaoLuo** [comment]
Interesting, I never knew the launchsettings.json overrides the environment variables currently in the system but it is documented here: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-2.1. So I assume this is "as designed"? Though I do think this is an odd default behaviour.

**Comment by Tratcher** [comment]
Yes, that sounds like it's working as expected. It wants the server to start with the urls specified in the launchsettings and the easiest way to do that is to set ASPNETCORE_URLS. This is specific to dev environments.

**Comment by Tratcher** [comment]
Note that if you specified urls in any of the other supported ways (config file, command line, code, etc.) most of those would also override ASPNETCORE_URLS.

**Comment by mikeharder** [comment]
I don't have an opinion either way, but I do want to point out this is a user experience regression from 2.0.  In CLI 2.1.105, no `launchSettings.json` is generated by default, so setting ASPNETCORE_URLS works.

@DamianEdwards , @davidfowl , @muratg: Any opinions on whether we should change the currently 2.1 experience?

**Comment by Tratcher** [comment]
Note this logic is built into the dotnet run tooling, it's not any code from the aspnet org. This bug would need to be moved accordingly.

**Comment by muratg** [comment]
I assume we'd like to bring this one to shiproom?

**Comment by Tratcher** [comment]
Should this be moved to https://github.com/dotnet/cli first?

**Comment by Tratcher** [comment]
https://github.com/dotnet/cli/blob/d0780d81e571abfcdcd1ed2ab85f0cc4a155528e/src/dotnet/commands/dotnet-run/LaunchSettings/ProjectLaunchSettingsProvider.cs#L18-L21

**Comment by muratg** [comment]
@livarcocc could you take a quick look before I move it over?

**Comment by livarcocc** [comment]
@vijayrkn can you take a look? your team has maintained this code on the CLI so far.

**Comment by vijayrkn** [comment]
@mlorbetske - do you know who else have context on this?

**Comment by mlorbetske** [comment]
Bill would probably be the best person to take a look

Get Outlook for iOS<https://aka.ms/o0ukef>
_____________________________

**Comment by vijayrkn** [comment]
@BillHiebert

**Comment by BillHiebert** [comment]
I have no context on the code in dotnet run, but this seems by design to me as the applicationUrls section in launchsettings.json file will cause it to use those settings. I assume if this didn't repro before, it was because either there wasn't a launchsettings.json file, or the dotnet run code never honored the settings in it. I don't think reverting it is the right behavior.
